### PR TITLE
Add opensource conan components

### PR DESCRIPTION
**Pull request by Lucas-Vertrees-Bose**

### CODE REVIEW

File: Bose/conanfile.py [hunk 1]
```diff
@@ -14,10 +14,19 @@ class OpenSourceCurl(common.MakefilePackage):
...
-        self.run("cd ./Bose;./build-%s" % self.settings.arch)
+        zlib_folder = self.deps_cpp_info["opensource-zlib"].rootpath
+        nghttp_folder = self.deps_cpp_info["opensource-nghttp2"].rootpath
+        openssl_folder = self.deps_cpp_info["opensource-openssl"].rootpath
+        export_cmd = ("export OPENSOURCEZLIB=%s " 
+                      "export OPENSOURCENGHTTP=%s " 
+                      "export OPENSOURCEOPENSSL=%s" 
+                      % (zlib_folder, nghttp_folder, openssl_folder))
+
+        self.run("%s && cd ./Bose;./build-%s" % ( export_cmd, self.settings.arch ) )
```
Comment (Derek-Richardson-Bose) [inline Bose/conanfile.py]: should this be a ; not a && so they are run sequentially

Comment (Michael-Cook-Bose) [reply Bose/conanfile.py]: Both `;` and `&&` ensure sequentially.
The difference is `;` would ignore the failure of the `%s && cd./Bose` commands.

File: Bose/build-qc8017_32 [hunk 2]
```diff
@@ -19,9 +19,27 @@ fi
 cd Bose/ || exit
 
 hsp_dir=$(components get Riviera-Toolchain installed_location) || exit
-zlib_dir=$(components get opensource-zlib-$sdk installed_location) || exit
-nghttp2_dir=$(components get opensource-nghttp2-$sdk installed_location) || exit
-openssl_dir=$(components get opensource-openssl-$sdk installed_location) || exit
+
+if [[ -z "$OPENSOURCEZLIB" ]]; then
```
Comment (Michael-Cook-Bose) [inline Bose/build-qc8017_32]: FWIW, this is overly verbose.
All you need to do is:
```suggestion
if [ "$OPENSOURCEZLIB" ]; then
```
A well-established idiom used far and wide.

File: Bose/conanfile.py [hunk 3]
```diff
@@ -14,10 +14,19 @@ class OpenSourceCurl(common.MakefilePackage):
...
-        self.run("cd ./Bose;./build-%s" % self.settings.arch)
+        zlib_folder = self.deps_cpp_info["opensource-zlib"].rootpath
+        nghttp_folder = self.deps_cpp_info["opensource-nghttp2"].rootpath
+        openssl_folder = self.deps_cpp_info["opensource-openssl"].rootpath
+        export_cmd = ("export OPENSOURCEZLIB=%s " 
+                      "export OPENSOURCENGHTTP=%s " 
+                      "export OPENSOURCEOPENSSL=%s" 
+                      % (zlib_folder, nghttp_folder, openssl_folder))
+
+        self.run("%s; cd ./Bose; ./build-%s" % ( export_cmd, self.settings.arch ) )
```
Comment (Michael-Cook-Bose) [inline Bose/conanfile.py]: I don't think this code is right.
export_cmd is a tuple of three strings.
So we're going to end up with a weird command like
```
export OPENSOURCEZLIB=.../zlib export OPENSOURCENGHTTP=.../nghttp2 export OPENSOURCEOPENSSL=.../openssl; cd ./Bose; ./build-foo
```
with extra "export" words in there.
Print the command before invoking `self.run` and you'll see what I mean.

I'd suggest just getting rid of the export_cmd variable. It only seems to complicate things.
```
self.run("""\
export OPENSOURCEZLIB={zlib} OPENSOURCENGHTTP={nghttp} OPENSOURCEOPENSSL={openssl}
cd ./Bose || exit
./build-{arch}
```.format(zlib=self.deps_cpp_info["opensource-zlib"].rootpath,
    nghttp=self.deps_cpp_info["opensource-nghttp2"].rootpath,
  openssl=self.deps_cpp_info["opensource-openssl"].rootpath,
  arch=self.settings.arch)

**Comment by Derek-Richardson-Bose** [comment]
@Michael-Cook-Bose  can we merge?

**Pull request merged**